### ISSUE TITLE
Implement "db.collection.aggregate"

### ISSUE THREAD

**Issue by pstromberg98** [williamkapke/mongo-mock#76]
The collection aggregate function needs implementing and is a pretty big feature. I am creating this issue to discuss implementation and to possibly split work amongst a couple of other contributors. I will start on implementation and continually add posts on any updates when appropriate.

Mongo Reference: https://docs.mongodb.com/manual/reference/method/db.collection.aggregate/

**Comment by AngefloMusic** [comment]
I'd be interested in working on this, but let me get a contribution guide up today first, this is a good example of where we'll need a feature branch to merge into, since several people can work on this in parallel.

**Comment by Pidid** [comment]
Sounds good. Yeah the merge policy is going to work well for this project!

**Comment by AngefloMusic** [comment]
Let's make a feature checklist for aggregation, these below from my experience are the most used of the aggregation commands:

- [ ] $match
- [ ] $project
- [ ] $group
- [ ] $unwind
- [ ] $sort
- [ ] $limit
- [ ] $count
- [ ] $lookup

**Comment by zardilior** [comment]
I can help if need be, is there anything I should now? I can do the basic match and group!!

**Comment by VikasChandrashekar** [comment]
I can help if you still need any help with this

**Comment by zardilior** [comment]
Guess we should just go ahead and do the PR

**Comment by DeekshaPandit** [comment]
Any update on this? Can I help?

**Comment by danmilgram** [comment]
Hello, any updates?
